api/app/Console/Commands/DepartmentDeduplication.php

```diff
@@ -0,0 +1,212 @@
+                'keep_number' => 19,
+                'remove_number' => 200,
+            ],
+            [
+                'name_en' => 'Registrar of the Supreme Court of Canada',
+                'keep_number' => 80,
+                'remove_number' => 214,
+            ],
+            [
+                'name_en' => 'Canada Border Services Agency',
+                'keep_number' => 85,
+                'remove_number' => 38,
+            ],
+            [
+                'name_en' => 'Canada Revenue Agency',
+                'keep_number' => 130,
+                'remove_number' => 122,
+            ],
+        ];
+
+        // run through each array in the map
+        foreach ($departmentMap as $departmentMapObject) {
+
+            /** @var \App\Models\Department | null $departmentToKeep */
+            $departmentToKeep = Department::where('department_number', $departmentMapObject['keep_number'])->first();
+            /** @var \App\Models\Department | null $departmentToRemove */
+            $departmentToRemove = Department::where('department_number', $departmentMapObject['remove_number'])->first();
+
+            // short circuit if department(s) not found
+            if ($departmentToKeep === null || $departmentToRemove === null) {
+                $this->info("Department name: {$departmentMapObject['name_en']} was SKIPPED");
+                $this->info('Department to keep or department to remove not found');
+
+                continue;
+            }
+
+            $updateCounter = 0;
+
+            //
+            // ELOQUENT SECTION
+            //
+
+            // Pools
+            $pools = Pool::where('department_id', $departmentToRemove->id)->get();
+            /** @var \App\Models\Pool $pool */
+            foreach ($pools as $pool) {
+                $pool->department_id = $departmentToKeep->id;
+                $pool->save();
+                $updateCounter++;
+            }
+
+            // PoolCandidateSearchRequests
+            $poolCandidateSearchRequests = PoolCandidateSearchRequest::where('department_id', $departmentToRemove->id)->get();
+            /** @var \App\Models\PoolCandidateSearchRequest $poolCandidateSearchRequest */
+            foreach ($poolCandidateSearchRequests as $poolCandidateSearchRequest) {
+                $poolCandidateSearchRequest->department_id = $departmentToKeep->id;
+                $poolCandidateSearchRequest->save();
+                $updateCounter++;
+            }
+
+            // PoolCandidates
+            $poolCandidates = PoolCandidate::where('placed_department_id', $departmentToRemove->id)->get();
+            /** @var \App\Models\PoolCandidate $poolCandidate */
+            foreach ($poolCandidates as $poolCandidate) {
+                $poolCandidate->placed_department_id = $departmentToKeep->id;
+                $poolCandidate->save();
+                $updateCounter++;
+            }
+
+            // team_department
+            // SKIP
+
```

Comment:
Table is empty for me locally
Is it relevant or should it have been deleted in https://github.com/GCTC-NTGC/gc-digital-talent/pull/13352 ?

Reply:
matt made a PR here https://github.com/GCTC-NTGC/gc-digital-talent/pull/13510

Reply:
sorry .. I took soo long for this.. this PR just teleported me to somewhere else and landed back just now 😆
Continuing this now ! Thanks for the great catch!